docs/proposals/tinkerbell-stack-crd.md

```diff
@@ -0,0 +1,172 @@
+# Tinkerbell Operator CRD
+
+
+## Overview
+Currently, the tinkerbell services are deployed by the operator using the manifest file (tinkerbell.yaml) found in the
+deploy directory during deployment. However, our goal is to deploy the tinkerbell services only after a
+CustomResourceDefinition (CRD) named Tinkerbell or TinkerbellInstance is created and to uninstall them
+when the CRD is removed.
+
+## Goals/Non-goals
+
+**Goals**
+
+- Define the CRD body and it's underlying specs
+- How the CRD influences the operators' behaviour(setting up TB services and cleaning them up)
+- Focus on Tinkerbell deployment specs and how they can be customized.
+
+**Non-goals**
+
+- Define a relationship between Tinkerbell core services or objects with this CRD
+- Address advanced use cases such as migrations and upgrades as this is too early at this stage
+- Extend the CRD spec to the `tink-stack` deployment as it is not yet clear how to handle it in the future
+- Customize Kubernetes native fields such as deployment resources, image pull policies, etc...
+
+## proposal
+
+### Custom Resource Definitions
+
+It is not yet clear about the name of the CRD at this point, however, I will use the name Tinkerbell but of course
+open for a change.
+
+
+#### Tinkerbell
```

Comment:
Could we get the intended API group also as it influences naming of the CRDs. My initial thought is `operator.tinkerbell.org` but that may be unnecessarily verbose. Could we have a short discussion with examples on what other operators do.

Reply:
Under this 'rationale' section, perhaps it would be useful to have a _short_ discussion about other operators and what they do?

Reply:
Yeah sure thing, I can add a Rationale section and we can add our thoughts in there. When it comes to the API group it doesn't have to be very explicit( as you said prefixing the api group with `operator` feels a bit verbose/explicit). Based on my experience with operators, the api groups or the CRs don't distinguish if this is an operator specific or not. Thus as an API group `tinkerbell.org` should be enough and the CR could be `tinkstack.tinkerbell.org`. 

As you said let's start a discussion about it, but in general I don't have a strong opinion about what I used/developed before so we can brainstorm together about :)